plugins/woocommerce/src/Admin/API/Patterns.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Admin\API;
+
+use Automattic\WooCommerce\Blocks\BlockPatterns;
+use Automattic\WooCommerce\Blocks\Package;
+use Automattic\WooCommerce\Blocks\Patterns\PTKClient;
+use Automattic\WooCommerce\Blocks\Patterns\PTKPatternsStore;
+use Automattic\WooCommerce\StoreApi\Exceptions\RouteException;
+use Exception;
+use WP_Error;
+use WP_HTTP_Response;
+use WP_REST_Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Product controller
+ *
+ * @internal
+ */
+class Patterns {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc-admin';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/patterns',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_pattern' ),
+					'permission_callback' => function () {
+						return is_user_logged_in();
+					},
+				),
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'update_patterns' ),
+					'permission_callback' => function () {
+						return is_user_logged_in();
+					},
+				),
+			)
+		);
+	}
+
+	/**
+	 * Fetch a single pattern from the PTK to ensure the API is available.
+	 *
+	 * @return WP_Error|WP_HTTP_Response|WP_REST_Response
```

Comment:
Is this accurate if it always returns `'success' => true` when it doesn't `throw` ? E.g. we never return `WP_Error`